src/psyclone/psyir/symbols/symbol_table.py

```diff
@@ -454,6 +460,15 @@ def find_or_create_tag(self, tag, root_name=None, **new_symbol_args):
         except KeyError:
             if not root_name:
                 root_name = tag
+            if exact_name:
```

Comment:
In new_symbol,  can you also add the check (or import-renaming logic) that I mentioned in https://github.com/stfc/PSyclone/issues/2537#issuecomment-2042818518

Reply:
Also make it fail for UnsupportedType which should not support renaming.